src/apps/fuzzers/fuzzerCompact.c

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2021 Uber Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/** @file
+ * @brief Fuzzer program for compactCells and uncompactCells
+ */
+
+#include "aflHarness.h"
+#include "h3api.h"
+#include "utility.h"
+
+const int MAX_UNCOMPACT_RES = 9;
+const int64_t MAX_UNCOMPACT_SIZE = 4000000;
+
+int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
+    if (size < sizeof(H3Index) * 2) {
+        return 0;
+    }
+
+    uint8_t res = *data;
+    uint8_t uncompactRes = (*(data + 1)) % (MAX_UNCOMPACT_RES + 1);
+    H3Index *input = (H3Index *)(data + sizeof(H3Index));
+    size_t inputSize = (size - sizeof(H3Index)) / sizeof(H3Index);
+
+    // fuzz compactCells
+    H3Index *compacted = calloc(inputSize, sizeof(H3Index));
+    H3Error compactErr = H3_EXPORT(compactCells)(input, compacted, inputSize);
+
+    // fuzz uncompactCells using output of above
+    if (!compactErr) {
+        int compactedCount = 0;
+        for (int i = 0; i < inputSize; i++) {
+            if (compacted[i] != H3_NULL) {
+                compactedCount++;
+            }
+        }
+        if (compactedCount < 2) {
+            int64_t uncompactedSize;
+            H3Error err = H3_EXPORT(uncompactCellsSize)(
+                compacted, inputSize, uncompactRes, &uncompactedSize);
+            if (!err && uncompactedSize < MAX_UNCOMPACT_SIZE) {
+                H3Index *uncompacted = calloc(uncompactedSize, sizeof(H3Index));
+                H3_EXPORT(uncompactCells)
+                (compacted, compactedCount, uncompacted, uncompactedSize,
```

Comment:
I don't think you can use `compactedCount` here, as nothing guarantees the non `H3_NULL` cells that you counted are at the front of the array.

Reply:
In that case I would suggest just removing this entire block from lines 42-61 since we test uncompact later on the original fuzzer input anyways.

Reply:
At least, I don't think so. But maybe we should have @dfellis confirm.

@isaacbrodsky, what was the reasoning for restricting to `< 2` compacted cells?

Reply:
As I recall I copied this from oss-fuzz, I think the idea there was the input was expected to behave in a certain way.